docs/src/content/docs/reference/main/data-types/temporal-types.adoc

```diff
@@ -0,0 +1,143 @@
+---
+title: XTDB Temporal Types
+---
+
+NOTE: The contents of this page focus mainly on construction and casting of Temporal Types. For information on functions which apply to temporal types, see the link:../stdlib/temporal[Temporal Functions] page.
+
+This document outlines the supported types of XTDB, alongside their usage, construction and casting behaviors.
+
+== Supported Temporal Types
+
+[cols="3,5",options="header"]
+|===
+|XTDB Type |Description
+
+|`TIMESTAMP [WITHOUT TIMEZONE]`
+|Represents both date and time *without* a time zone offset.
+
+|`TIMESTAMP WITH TIMEZONE`
+|Represents both date and time *with* a time zone offset.
+
+|`DATE`
+|Represents a date without a time component.
+
+|`TIME [WITHOUT TIME ZONE]`
+|Represents the time of day without a date.
+
+|`INTERVAL [INTERVAL QUALIFIER] [(p)]`
+|Represents a span of time without a specific start or end timestamp. 
+
+|`PERIOD`
+|Represents a specific start and end timestamp, capturing a range of time.
+
+|`DURATION`
+|Represents duration of time - purely in hours, minutes, seconds and fractional seconds.
+|===
+
+
+== Temporal Constructors and Literals
+
+Temporal values are instantiated using syntax compliant with SQL standards or ISO8601 strings for broader compatibility.
+
+=== SQL Examples
+
+[source,sql]
+----
+DATE '2023-03-15' -- Constructs a date.
+TIMESTAMP '2023-03-15 12:00:00' -- Timestamp without time zone.
+TIMESTAMP '2023-03-15T12:00:00' -- Timestamp without time zone in ISO8601 format.
+TIMESTAMP '2023-03-15 12:00:00+01:00' -- Timestamp with time zone.
+TIME '14:30:00' -- Time without time zone.
+INTERVAL '1-2' YEAR TO MONTH -- A Year-Month Interval specified with an interval qualifier.
+INTERVAL 'P1Y2M' -- The Same Interval specified in a ISO8601 format.
+INTERVAL '1 2:30:10.111111' DAY TO SECOND(6) -- A Day-Time Interval specified with an interval qualifier.
+INTERVAL 'P1DT2H30M10.111111S' -- The Same Interval specified in a ISO8601 format.
+DURATION 'P20H3M11.1S' -- A Duration specified in ISO8601 format.
+PERIOD(DATE '1998-01-05', DATE '1998-01-12') -- A Period specified with two dates.
+----
+
+=== XTQL/Clojure Examples
```

Comment:
We might be better to have XTQL/JSON examples here instead, or as well, but ahead of the XTQL/Clojure. Or perhaps both should live under the various "Drivers" sections (might be a good opporunity to use asciidoc Includes?)

Reply:
Current thoughts/changes might make here:
- Move Clojure examples out to a separate document "temporal types".
- Create an issue around missing XTQL JSON types.
- Keep SQL examples on the page - and maybe add JSON as/when they're ready to go.

Can merge what we have afterwards as it should be an improvement on what we already have :)